### PR TITLE
AI will now found & enhance religions -- minor improvement to civilian AI

### PR DESCRIPTION
**Pull request by xlenstra**
This PR adds simple functionality for AI players to found and enhance religions.
It also gives an AI for prophets to use their 'found religion' and 'enhance religion' actions, though no other actions are supported as of yet, most notably 'spread religion'.
The beliefs for these are chosen randomly, I'll port the AI from the original in another PR.

I've also made some changes to automated civilian movements, mostly that if there is danger nearby they will always try to flee to either a city or a military unit, hopefully leading to less of them being destroyed and/or captured.

### CODE REVIEW

File: core/src/com/unciv/logic/automation/SpecificUnitAutomation.kt [hunk 1]
```diff
@@ -369,4 +369,27 @@ object SpecificUnitAutomation {
...
         return false
     }
+
+    fun foundReligion(unit: MapUnit) {
+        val cityToFoundReligionAt = unit.civInfo.cities.first { !it.isHolyCity() }
+        if (unit.getTile() != cityToFoundReligionAt.getCenterTile()) {
+            unit.movement.headTowards(cityToFoundReligionAt.getCenterTile())
+            return   
+        }
+        unit.civInfo.religionManager.useGreatProphet(unit)
```
Comment (yairm210) [inline core/src/com/unciv/logic/automation/SpecificUnitAutomation.kt:379]: This should be "use the unit action" instead, so the logic is only in one place

File: core/src/com/unciv/logic/automation/UnitAutomation.kt [hunk 2]
```diff
@@ -469,16 +497,43 @@ object UnitAutomation {
         unit.civInfo.addNotification("[${unit.displayName()}] finished exploring.", unit.currentTile.position, unit.name, "OtherIcons/Sleep")
         unit.action = null
     }
-
+    
+    /** Returns whether the civilian spends its turn hiding and not moving */
+    fun tryRunAwayIfNeccessary(unit: MapUnit): Boolean {        
+        // This is a little 'Bugblatter Beast of Traal': Run if we can attack an enemy
```
Comment (yairm210) [inline core/src/com/unciv/logic/automation/UnitAutomation.kt:503]: "... A creature so mondbogglingly stupid..."

File: core/src/com/unciv/logic/civilization/ReligionManager.kt [hunk 3]
```diff
@@ -137,9 +137,11 @@ class ReligionManager {
             if (!mayFoundReligionNow(prophet)) return // How did you do this?
             religionState = ReligionState.FoundingReligion
```
Comment (yairm210) [inline core/src/com/unciv/logic/civilization/ReligionManager.kt:138]: I think this whole function should be put into the unit actions class

File: core/src/com/unciv/logic/automation/UnitAutomation.kt (hunk 2 above)
Comment (xlenstra) [reply core/src/com/unciv/logic/automation/UnitAutomation.kt:503]: Not my comment - just copied and pasted it from somewhere :)

Comment (SomeTroglodyte) [reply core/src/com/unciv/logic/automation/UnitAutomation.kt:503]: Finally someone noticed. Had to add those to the dictionary, too, so Marvin Studio stopped squiggying them!

Comment (SomeTroglodyte) [reply core/src/com/unciv/logic/automation/UnitAutomation.kt:503]: But honestly, that should have been a to-do: The check _is_ the wrong way round, at least where I put the comment.

**Comment by SomeTroglodyte** [comment]
Can't really check this in depth now, looks superficially like done with a concentrated mind. Only thing - I see a lot of code very specific to the Prophet _not_ be in SpecificUnitAutomation. But since you moved the 'run away' test from specific to the other file as it _is_ no longer specific I'll assume you chose function containers carefully enough...

A thought: It might be mod-friendlier to not 'switch' on the possible roles, but to try in turn, the specific subfunction returns success/nope, and if nope try another role - as per a discussion with @ravignir in... 5010? You've seen it, surely. Anyway, include and test that while you're at it? Or should we do a fuzzy evaluateWorthOfWhatICanDoInThisRole(): Float - sorted() - first() - action approach much later?

**Pull request merged**